api/PayrollProcessor.Functions.Api/Infrastructure/QueueMessageHandler.cs

```diff
@@ -0,0 +1,12 @@
+using Microsoft.WindowsAzure.Storage.Queue;
+using Newtonsoft.Json;
+using PayrollProcessor.Data.Persistence.Infrastructure.Clients;
+
+namespace PayrollProcessor.Functions.Api.Infrastructure
+{
+    public class QueueMessageHandler
+    {
+        public static TMessage FromQueueMessage<TMessage>(CloudQueueMessage message) where TMessage : IMessage =>
```

Comment:
Azure Functions uses `CloudQueueMessage`, whereas the latest Azure Queue library from Microsoft uses the type `QueueMessage`.

The Azure Functions bindings for Azure Storage are slightly out of date.

This means we can't share the same APIs/abstractions in the `Data.Persistence` library and the `Functions.Api` application.